test/integration/targets/aws_lambda/tasks/main.yml

```diff
@@ -0,0 +1,409 @@
+---
+#
+#  Author: Michael De La Rue
+#  based on ec2_key.yml + lambda.py
+
+- block:
+
+    # ============================================================
+    - name: test with no parameters
+      lambda:
+      register: result
+      ignore_errors: true
+
+    - name: assert failure when called with no parameters
+      assert:
+        that:
+           - 'result.failed'
+           - 'result.msg.startswith("missing required arguments: name")'
```

Comment:
Do we really need to check that `AnsibleModule(argument_spec=argument_spec)` does the right thing?

I'd be in favour of limiting the tests to checking things that the core code doesn't check. I guess someone could remove `required=True` from the `name` argument_spec in future and we might not notice.